File: 프로그래머스/2/42885. 구명보트/구명보트.py
from collections import deque

def solution(people, limit):
    answer = 0
    people.sort()
    people = deque(people)
    
    while people:
        if len(people) > 1 and limit - people[-1] >= people[0]:
            people.popleft()
            people.pop()
        else:
            people.pop()
        answer += 1
        
    return answer

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        int start = 0;
        int end = people.length - 1;
        Arrays.sort(people);
        
        while (start <= end) {
            if (people[end] + people[start] <= limit) {
                start++;
            }
            end--;
            answer++;
        }
        return answer;
    }
}